pkg/apis/apps/v1beta2/conversion_test.go

```diff
@@ -0,0 +1,561 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2_test
+
+import (
+	"testing"
+
+	"k8s.io/api/apps/v1beta2"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestV1beta2StatefulSetSpecConversion(t *testing.T) {
+	replicas := newInt32(2)
+	selector := &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}
+	v1beta2Template := v1.PodTemplateSpec{ObjectMeta: metav1.ObjectMeta{Name: "foo"}, Spec: v1.PodSpec{RestartPolicy: v1.RestartPolicy("bar")}}
+	apiTemplate := api.PodTemplateSpec{ObjectMeta: metav1.ObjectMeta{Name: "foo"}, Spec: api.PodSpec{RestartPolicy: api.RestartPolicy("bar")}}
+	testcases := map[string]struct {
+		stsSpec1   *apps.StatefulSetSpec
+		stsSepc2   *v1beta2.StatefulSetSpec
+		adjustment bool
+	}{
+		"StatefulSetSpec Conversion 1": {
+			stsSpec1:   &apps.StatefulSetSpec{Replicas: *replicas, Template: apiTemplate},
+			stsSepc2:   &v1beta2.StatefulSetSpec{Replicas: replicas, Template: v1beta2Template},
+			adjustment: true,
+		},
+		"StatefulSetSpec Conversion 2": {
+			stsSpec1:   &apps.StatefulSetSpec{Replicas: *replicas, Selector: selector, Template: apiTemplate, ServiceName: "foo", PodManagementPolicy: apps.PodManagementPolicyType("bar")},
+			stsSepc2:   &v1beta2.StatefulSetSpec{Replicas: replicas, Selector: selector, Template: v1beta2Template, ServiceName: "foo", PodManagementPolicy: v1beta2.PodManagementPolicyType("bar")},
+			adjustment: true,
+		},
+	}
+
+	for k, tc := range testcases {
+		// apps -> v1beta2
+		internal1 := &v1beta2.StatefulSetSpec{}
+		if err := api.Scheme.Convert(tc.stsSpec1, internal1, nil); err != nil {
+			t.Errorf("%q - %q: unexpected error: %v", k, "from extensions to v1beta2", err)
+		}
+
+		if !apiequality.Semantic.DeepDerivative(internal1, tc.stsSepc2) {
```

Comment:
Use `apiequality.Semantic.DeepEqual` instead of DeepDerivative everywhere in this PR.

Reply:
@janetkuo Updated. PTAL. Thanks.